app/main.py

```diff
@@ -1 +1,33 @@
-# write your code here
+import datetime
+from app.cafe import Cafe
+from app.errors import (
+    VaccineError,
+    OutdatedVaccineError,
+    NotWearingMaskError
+)
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
+    friend_with_no_mask = 0
+    try:
+        for friend in friends:
+            if not friend.get("vaccine"):
+                raise OutdatedVaccineError
+            elif friend["vaccine"]["expiration_date"] < datetime.date.today():
+                raise OutdatedVaccineError
+
+        for friend in friends:
+            if friend["wearing_a_mask"] is False:
+                friend_with_no_mask += 1
+
+        if friend_with_no_mask:
+            raise NotWearingMaskError(friend_with_no_mask)
```

Comment:
There is no need to perform these checks here, relay only on the `visit_cafe` method, it will raise necessary errors. Remove the code from lines 13 to 24 to not violate the DRY.